feat: Add 'Try in Web IDE' GitHub action blog post (#6)
eclipse-che/blog: _posts/2021-10-14-test-pull-requests-with-the-try-in-web-ide-github-action.md

---
title: Test pull requests with the Try in Web IDE GitHub action
layout: post
author: David Kwon
description: >-
  How to integrate the Try in Web IDE GitHub action to your GitHhub repository, making it easier for collaborators to test pull requests in Red Hat CodeReady Workspaces.
categories: []
keywords: ['GitHub', 'action', 'pull request', 'try', 'in', 'web', 'IDE', 'browser', 'repository']
slug: /@david.kwon/test-pull-requests-with-the-try-in-web-ide-github-action
---

*Cross posted from the [Red Hat Developer blog](https://developers.redhat.com/articles/2021/10/07/test-pull-requests-try-web-ide-github-action#integrate_the_try_in_web_ide_action_with_your_github_repository).*

> Web-based, cloud-hosted integrated development environments (IDEs) can make development and collaboration more efficient if they are well integrated with version control. This blog post shows you how to use the [Try in Web IDE GitHub action](https://github.com/marketplace/actions/try-in-web-ide) to make it easier for collaborators to test pull requests on [Red Hat CodeReady Workspaces](https://developers.redhat.com/products/codeready-workspaces/overview) running in the [Developer Sandbox for Red Hat OpenShift](https://developers.redhat.com/developer-sandbox/ide).

Reviewer: ```suggestion
Web-based, cloud-hosted integrated development environments (IDEs) can make development and collaboration more efficient if they are well integrated with version control. This blog post shows you how to use the [Try in Web IDE GitHub action](https://github.com/marketplace/actions/try-in-web-ide) to make it easier for collaborators to test pull requests on [Eclipse Che hosted by Red Hat](https://www.eclipse.org/che/docs/che-7/hosted-che/hosted-che/).
```

> Note: CodeReady Workspaces is Red Hat's supported version of Eclipse Che.

Reviewer: ```suggestion
Note: The instance of Eclipse Che used by Try in Web IDE GitHub action can be configured. The default one is  [workspaces.openshift.com](https://workspaces.openshift.com) (Eclipse Che hosted by Red Hat). 
```

The [Eclipse Che documentation](https://github.com/eclipse/che-docs) repository has recently integrated this GitHub action into its workflow. You can view a recent pull request to try it yourself. Figure 1 shows a pull request comment created by the GitHub action. Clicking the badge opens a new workspace to try the pull request in the web browser.

![GitHub PR comment with badge to open PR branch in new workspace]({{ site.url }}{{ site.baseurl }}/assets/img/test-pull-requests-with-the-try-in-web-ide-github-action/comment.png)
<span style="color:grey">*Figure 1: Clicking the badge opens a web IDE workspace for testing the pull request.*</span>

Figure 2 shows a status check done by the GitHub action. Clicking the Details link opens a new workspace to try the pull request in the web browser.

![GitHub PR status check with link to open PR branch in new workspace]({{ site.url }}{{ site.baseurl }}/assets/img/test-pull-requests-with-the-try-in-web-ide-github-action/status-check.png)
<span style="color:grey">*Figure 2: Clicking the Details link opens a web IDE workspace for testing the pull request.*</span>

> Figure 3 shows the workspace created in CodeReady Workspaces in the Developer Sandbox. This is the web IDE workspace that opens in the web browser when you click either the badge from Figure 1 or the link from Figure 2. From here, you can try the pull request and test its correctness.

Reviewer: ```suggestion
Figure 3 shows the workspace created in Eclipse Che hosted by Red Hat. This is the web IDE workspace that opens in the web browser when you click either the badge from Figure 1 or the link from Figure 2. From here, you can try the pull request and test its correctness.
```

![PR branch opened in a new workspace]({{ site.url }}{{ site.baseurl }}/assets/img/test-pull-requests-with-the-try-in-web-ide-github-action/crw.png)
<span style="color:grey">*Figure 3: Try the pull request and test its correctness.*</span>

## Integrate the Try in Web IDE action with your GitHub repository
This section shows you how to add the Try in Web IDE GitHub action to your GitHub repository's workflow. We will configure the action to automatically create a comment (Figure 1) and status check (Figure 2) on new pull requests.

### Prerequisites and setup

> You will need a Red Hat account to use CodeReady Workspaces in the Developer Sandbox. Navigate to [Developer Sandbox for Red Hat OpenShift](https://developers.redhat.com/developer-sandbox), register for a free account, and launch your Developer Sandbox environment. Note that you must verify your identity with a phone number.

Reviewer: ```suggestion
You will need a Red Hat account to use Eclipse Che hosted by Red Hat. Navigate to [Developer Sandbox for Red Hat OpenShift](https://developers.redhat.com/developer-sandbox), register for a free account, and launch your Developer Sandbox environment. Note that you must verify your identity with a phone number.
```

You also need an account on GitHub and a repository where you can integrate GitHub actions.

> Finally, you should have a [devfile](https://www.eclipse.org/che/docs/che-7/end-user-guide/authoring-devfiles-version-1/) in the root of your GitHub repository. We'll use the devfile very shortly.

Reviewer: ```suggestion
Finally, although not mandatory, you can add a [devfile](https://www.eclipse.org/che/docs/che-7/end-user-guide/authoring-devfiles-version-1/) in the root of your GitHub repository to specify the features of the development environment and IDE. We'll use the devfile very shortly.
```

You can further configure the workflow defined in `example.yml` to fit your needs. Consider adding more activity types within the `on.pull_request_target.types` field. Alongside the `opened` event, other events that you might find useful are `reopened` (which is triggered when the pull request is reopened) and `synchronize` (which is triggered when the pull request's tracking branch synchronizes with its source branch). The new types are added in the following code snippet:
```yaml
...

on:
  pull_request_target:
    # Add multiple activity types
    types: [opened, reopened, synchronize]

...
```

The `add_comment` and `add_status` GitHub action inputs can also be configured to customize whether the comment or status check is created in the pull request. For example, the following code snippet disables pull request comments:
```yaml
        ...

        with:
          {% raw %}github_token: ${{ secrets.GITHUB_TOKEN }}{% endraw %}
          add_comment: false
          add_status: true
```
The full table of inputs is available in the Try in Web IDE GitHub action [documentation](https://github.com/marketplace/actions/try-in-web-ide#action-inputs).

> ### Step 3: Create a devfile

Reviewer: ```suggestion
### Step 3 (optional): Create a devfile
```
